OpenEdXMobile/src/main/java/org/edx/mobile/notifications/services/NotificationService.java

```diff
@@ -0,0 +1,80 @@
+package org.edx.mobile.notifications.services;
+
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.support.v4.app.NotificationCompat;
+import android.support.v4.app.TaskStackBuilder;
+import android.util.Log;
+
+import com.google.firebase.messaging.FirebaseMessaging;
+import com.google.firebase.messaging.FirebaseMessagingService;
+import com.google.firebase.messaging.RemoteMessage;
+
+import org.edx.mobile.R;
+import org.edx.mobile.base.MainApplication;
+import org.edx.mobile.core.IEdxEnvironment;
+import org.edx.mobile.view.MainDashboardActivity;
+
+
+public class NotificationService extends FirebaseMessagingService {
+    public static final String NOTIFICATION_TOPIC_RELEASE = "edx_release_notification";
+
+    public static void subscribeToTopics(IEdxEnvironment environment){
+        if(environment.getConfig().areNotificationsEnabled()) {
+            FirebaseMessaging.getInstance().subscribeToTopic(
+                    NotificationService.NOTIFICATION_TOPIC_RELEASE
+            );
+        }
+    }
+
+    @Override
+    public void onMessageReceived(RemoteMessage remoteMessage) {
+        IEdxEnvironment environment = MainApplication.getEnvironment(this);
+
+        if(!environment.getConfig().areNotificationsEnabled()){
+            // Do not process Notifications when they are disabled.
+            return;
+        }
+
+        super.onMessageReceived(remoteMessage);
+
+        if (remoteMessage.getNotification() != null) {
+            Log.d(this.getClass().getSimpleName(),
+                    "Message Notification Body: " + remoteMessage.getNotification().getBody());
+        }
+
+        // Build out the Notification and set the intent to direct the user to the application
+        NotificationCompat.Builder builder =
+                new NotificationCompat.Builder(this)
+                        .setSmallIcon(R.drawable.ic_notification)
+                        .setContentTitle(remoteMessage.getNotification().getTitle())
+                        .setContentText(remoteMessage.getNotification().getBody());
+        Intent resultIntent = new Intent(this, MainDashboardActivity.class);
```

Comment:
I have concern over it, is it the right activity to launch for logged out users? 🤔

Reply:
I am updating this to the DiscoveryLaunchActivity, is that the best activity for logged in or logged out?  Or should I check if they are logged in and differentiate?

Reply:
IMO launcher activity i.e. 'SplashActivity' is good for now.

Reply:
@farhan Thank you, when we add Deep linking this will get more complicated.  I was also just running into issues with the Activity I was using.  

Let me try SplashActivity.